services/scabbard/libscabbard/src/service/state.rs

```diff
@@ -241,6 +241,8 @@ impl ScabbardState {
             .map_err(|_| ScabbardStateError("failed to receive result in reasonable time".into()))?
             .ok_or_else(|| ScabbardStateError("no result returned from executor".into()))?;
 
+        scheduler.shutdown();
```

Comment:
Couldn't any one of lines 239-242 still create the same problem?

Reply:
Good catch; fixed